ERROR:  Package is not installed: ext-xdebug-2.4.0.0

composer install throw exception

```
composer install
You are running composer with xdebug enabled. This has a major impact on runtime performance. See https://getcomposer.org/xdebug
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Package operations: 59 installs, 0 updates, 1 removal

                                                
  [InvalidArgumentException]                    
  Package is not installed: ext-xdebug-2.4.0.0  
                                                

install [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [--no-suggest] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--ignore-platform-reqs] [--] [<packages>]...

```

php version

```
php --version
PHP 5.6.22 (cli) (built: Jun 18 2016 16:47:14) 
Copyright (c) 1997-2016 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2016 Zend Technologies
    with Xdebug v2.4.0, Copyright (c) 2002-2016, by Derick Rethans
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2016, by Zend Technologies

```


composer version

```
composer --version
You are running composer with xdebug enabled. This has a major impact on runtime performance. See https://getcomposer.org/xdebug
Composer version 1.3.0 2016-12-24 00:47:03

```

if i use the previous stable version 1.2.4 of composer the install command works just fine

Yes it's present. (including Jenkins console log output for 1.3)
```
[exec] You are running composer with xdebug enabled. This has a major impact on runtime performance. See https://getcomposer.org/xdebug
[exec] Loading composer repositories with package information
[exec] Installing dependencies (including require-dev) from lock file
[exec] Package operations: 64 installs, 0 updates, 1 removal
[exec] 
[exec]                                                 
[exec]   [InvalidArgumentException]                    
[exec]   Package is not installed: ext-xdebug-2.5.0.0  
[exec]                                                 
[exec] 
[exec] install [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [--no-suggest] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--ignore-platform-reqs] [--] [<packages>]...
[exec] 

BUILD FAILED
```

--- a/src/Composer/XdebugHandler.php
+++ b/src/Composer/XdebugHandler.php
@@ -79,6 +79,11 @@ class XdebugHandler
                     putenv('PHP_INI_SCAN_DIR');
                 }
             }
+
+            // Clear version if the restart failed to disable xdebug
+            if ($this->loaded) {
+                putenv(self::ENV_VERSION);
+            }
         }
     }
 

--- a/tests/Composer/Test/XdebugHandlerTest.php
+++ b/tests/Composer/Test/XdebugHandlerTest.php
@@ -113,12 +113,31 @@ class XdebugHandlerTest extends \PHPUnit_Framework_TestCase
         $xdebug = new XdebugHandlerMock($loaded);
         $xdebug->check();
         $this->assertEquals($xdebug->testVersion, getenv(XdebugHandlerMock::ENV_VERSION));
+
+        // Mimic successful restart
+        $loaded = false;
+        $xdebug = new XdebugHandlerMock($loaded);
+        $xdebug->check();
+        $this->assertEquals($xdebug->testVersion, getenv(XdebugHandlerMock::ENV_VERSION));
     }
 
     public function testEnvVersionWhenNotLoaded()
     {
         $loaded = false;
 
+        $xdebug = new XdebugHandlerMock($loaded);
+        $xdebug->check();
+        $this->assertEquals(false, getenv(XdebugHandlerMock::ENV_VERSION));
+    }
+
+    public function testEnvVersionWhenRestartFails()
+    {
+        $loaded = true;
+
+        $xdebug = new XdebugHandlerMock($loaded);
+        $xdebug->check();
+
+        // Mimic failed restart
         $xdebug = new XdebugHandlerMock($loaded);
         $xdebug->check();
         $this->assertEquals(false, getenv(XdebugHandlerMock::ENV_VERSION));

